lib/libzfs/libzfs_dataset.c

```diff
@@ -196,6 +196,16 @@ zfs_validate_name(libzfs_handle_t *hdl, const char *path, int type,
 				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
 				    "reserved disk name"));
 				break;
+			
+			case NAME_ERR_SELF_MOUNT:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "self reference is found"));
+				break;
+			
+			case NAME_ERR_PARENT_MOUNT:
+				zfs_error_aux(hdl, dgettext(TEXT_DOMAIN,
+				    "parent path is found"));
```

Comment:
To be more clear, perhaps "filesystem name references parent"?

Reply:
How about NAME_ERR_PARENT_REFERENCE